internal/oidc/callback/callback_handler.go

```diff
@@ -301,9 +325,7 @@ func makeDownstreamSession(subject string, username string, groups interface{})
 	}
 	openIDSession.Claims.Extra = map[string]interface{}{
 		oidc.DownstreamUsernameClaim: username,
-	}
-	if groups != nil {
-		openIDSession.Claims.Extra[oidc.DownstreamGroupsClaim] = groups
+		oidc.DownstreamGroupsClaim:   groups,
```

Comment:
Instead of making `getGroupsFromUpstreamIDToken` always return a `[]string{}`, I would do that logic here and just return `nil` in `getGroupsFromUpstreamIDToken`.